internal/host/validator.go

```diff
@@ -1829,3 +1831,62 @@ func (v *validator) inventoryHasIP(inventory *models.Inventory, ipAddress string
 	}
 	return false, nil
 }
+
+func (v *validator) iSCSIHostNetworkInterfaceDoesNotBelongToMachineCidr(c *validationContext) (ValidationStatus, string) {
+	installationDisk, err := hostutil.GetHostInstallationDisk(c.host)
+	if err != nil || installationDisk == nil {
+		return ValidationSuccessSuppressOutput, ""
+	}
+
+	if installationDisk.DriveType != models.DriveTypeISCSI {
+		// Validation is not relevant in this case
+		return ValidationSuccessSuppressOutput, ""
+	}
+
+	if c.inventory == nil || !network.IsMachineCidrAvailable(c.cluster) {
+		return ValidationPending, "Missing inventory or machine network CIDR"
+	}
+
+	if installationDisk.Iscsi == nil {
+		// If this is nil, the disk shouldn't have passed the eligilibily test in the first place
+		v.log.Warn("iSCSI installation disk is missing host IP address")
+		return ValidationError, "iSCSI installation disk is missing host IP address"
+	}
+
+	nic, err := network.FindInterfaceByIPString(installationDisk.Iscsi.HostIPAddress, c.inventory.Interfaces)
+	if err != nil {
+		v.log.WithError(err).Warn("Cannot find network interface associated to iSCSI host IP address")
+		return ValidationError, "Cannot find network interface associated to iSCSI host IP address"
+	}
+
+	// Look if one of the IP set on the interface connected to the iSCSI disk belong to machine network CIDRs
+	ips := append(nic.IPV4Addresses, nic.IPV6Addresses...)
+	_, found := lo.Find(ips, func(ip string) bool {
+		var nicIP netip.Prefix
+		nicIP, err = netip.ParsePrefix(ip)
+		if err != nil {
+			return true
+		}
+
+		_, found := lo.Find(c.cluster.MachineNetworks, func(machineNetwork *models.MachineNetwork) bool {
+			var machineNetworkPrefix netip.Prefix
+			machineNetworkPrefix, err = netip.ParsePrefix(string(machineNetwork.Cidr))
+			if err != nil {
+				return true
+			}
+			return machineNetworkPrefix.Contains(nicIP.Addr())
+		})
+		return found
+	})
```

Comment:
If I understand, you are trying to check if the host IP in the CIDR
let take this part and this check 
https://github.com/openshift/assisted-service/blob/master/internal/network/machine_network_cidr.go#L519
and create a common function to serve them both

make less repeating code